rust/private/rustc.bzl

```diff
@@ -115,15 +115,15 @@ def collect_deps(deps, toolchain):
 
     crate_list = transitive_crates.to_list()
     transitive_libs = depset([c.output for c in crate_list]) + transitive_staticlibs + transitive_dylibs
-    indirect_crates = depset([crate for crate in crate_list if crate not in direct_crates])
+    indirect_crates = depset([crate for crate in crate_list if crate not in direct_crates.to_list()])
```

Comment:
I just realize that we flatten depset everywhere here. Maybe we should take a deeper look at performance implication of the DepInfo. /cc @mfarrugi @acmcarther

Reply:
I've added some TODOs

Reply:
Awesome, thanks!

Reply:
Of the [usages](https://github.com/bazelbuild/rules_rust/search?q=to_list&unscoped_q=to_list) only https://github.com/bazelbuild/rules_rust/blob/60c362762f2112b4366edc7eeea79ea3122bbae2/rust/private/rustc.bzl#L122 is easy to remove.

I think https://github.com/bazelbuild/rules_rust/blob/60c362762f2112b4366edc7eeea79ea3122bbae2/rust/private/rustdoc_test.bzl#L67-L74 could use Args, but I couldn't figure out how to turn it into a string in the template 🤷‍♂️